syntax = "proto3";

import "bob_emploi/frontend/api/geo.proto";
import "bob_emploi/frontend/api/job.proto";

package bayes.bob.old;

message JobsExploration {
  // A sorted list of job groups to explore. We believe that the first are the
  // more interesting to the user.
  repeated JobGroupExploration job_groups = 1;

  // The job that is used as the source of the exploration.
  Job source_job = 2;

  // The city that is used as the context of the exploration.
  FrenchCity city = 3;
}

message JobGroupExploration {
  // Description of the job group to explore.
  JobGroup job_group = 1;

  // Statistics about this job group in the user's context.
  LocalJobStats stats = 2;

  // A score to rank this among other job groups to explore.
  int32 score = 3;

  // A set of links to external website with more info to explore.
  repeated ExternalExplorationLink links = 4;

  // Type of mobility between two job groups.
  MobilityType mobility_type = 5;
}

message ExternalExplorationLink {
  // The caption to show when displaying the link.
  string caption = 1;

  // The URL to the external site.
  string url = 2;
}

// Request for exploration info around a job for a given city.
message JobExplorationRequest {
  Job source_job = 1;

  FrenchCity city = 2;
}

enum MobilityType {
  UNKNOWN_MOBILITY_TYPE = 0;

  // The job groups is close to the user's job group.
  CLOSE = 1;

  // The job group can be consider as an evolution from
  // the user's job group.
  EVOLUTION = 2;
}
